mpas_analysis/ocean/streamfunction_moc.py

```diff
@@ -250,44 +250,40 @@ def _load_mesh(self):  # {{{
             refTopDepth, refLayerThickness
         # }}}
 
-    def _cache_velocity_climatologies(self):  # {{{
+    def _compute_velocity_climatologies(self):  # {{{
         '''compute yearly velocity climatologies and cache them'''
 
-        variableList = ['avgNormalVelocity',
-                        'avgVertVelocityTop']
+        variableList = ['timeMonthly_avg_normalVelocity',
+                        'timeMonthly_avg_vertVelocityTop']
 
         config = self.config
 
-        outputDirectory = build_config_full_path(config, 'output',
-                                                 'mpasClimatologySubdirectory')
+        outputRoot = build_config_full_path(config, 'output',
+                                            'mpasClimatologySubdirectory')
 
-        make_directories(outputDirectory)
+        outputDirectory = '{}/meanVelocity'.format(outputRoot)
 
-        chunking = config.getExpression(self.sectionName, 'maxChunkSize')
-        ds = open_multifile_dataset(
-            fileNames=self.inputFilesClimo,
-            calendar=self.calendar,
-            config=config,
-            simulationStartTime=self.simulationStartTime,
-            timeVariableName='Time',
-            variableList=variableList,
-            variableMap=self.variableMap,
-            startDate=self.startDateClimo,
-            endDate=self.endDateClimo,
-            chunking=chunking)
+        changed, self.startYearClimo, self.endYearClimo = \
+            update_start_end_year_from_file_names(self.inputFilesClimo,
+                                                  config, self.calendar)
 
-        # update the start and end year in config based on the real extend of
-        # ds
-        update_start_end_year(ds, config, self.calendar)
-        self.startYearClimo = config.getint('climatology', 'startYear')
-        self.endYearClimo = config.getint('climatology', 'endYear')
+        self.velClimoFile = \
+            '{}/mpaso_ANN_{:04d}01_{:04d}12_climo.nc'.format(
+                    outputDirectory, self.startYearClimo,
+                    self.endYearClimo)
 
-        cachePrefix = '{}/meanVelocity'.format(outputDirectory)
+        if not os.path.exists(self.velClimoFile):
+            make_directories(outputDirectory)
 
-        # compute and cache the velocity climatology
-        cache_climatologies(ds, monthDictionary['ANN'],
-                            config, cachePrefix, self.calendar,
-                            printProgress=True)
+            compute_climatologies_with_ncclimo(
+                    config=config,
+                    inDirectory=self.historyDirectory,
+                    outDirectory=outputDirectory,
+                    startYear=self.startYearClimo,
+                    endYear=self.endYearClimo,
+                    variableList=variableList,
+                    modelName='mpaso',
+                    decemberMode='sdd')
```

Comment:
I do this in #222 but I don't think we need to do it here because the version of nco that supports the --seasons flag isn't available on conda-forge yet.  This way we can merge this PR anytime we want, whereas we need to wait for a new release of nco and an update of the ACME anaconda environments before we can widely deploy #222.